### ISSUE TITLE
Failed to fetch [com.nomnom.unity-project-patcher-bepinex] version

### ISSUE THREAD

**Issue by hoffor** [nomnomab/unity-project-patcher#4]
new HDRP project 2022.3.9f1
after 'add package from git url': `https://github.com/nomnomab/unity-project-patcher.git`
```
Failed to fetch [com.nomnom.unity-project-patcher-bepinex] version from "https://github.com/nomnomab/unity-project-patcher-bepinex".
UnityEngine.Debug:LogWarning (object)
Nomnom.UnityProjectPatcher.Editor.PatcherWindow/<GetPackageVersions>d__14:MoveNext () (at ./Library/PackageCache/com.nomnom.unity-project-patcher@db0e232801/Editor/PatcherWindow.cs:117)
System.Linq.Enumerable:ToArray<Nomnom.UnityProjectPatcher.Editor.PatcherWindow/PackageVersion> (System.Collections.Generic.IEnumerable`1<Nomnom.UnityProjectPatcher.Editor.PatcherWindow/PackageVersion>)
Nomnom.UnityProjectPatcher.Editor.PatcherWindow:OnLoad () (at ./Library/PackageCache/com.nomnom.unity-project-patcher@db0e232801/Editor/PatcherWindow.cs:35)
UnityEditor.EditorAssemblies:ProcessInitializeOnLoadMethodAttributes ()

```

**Comment by nomnomab** [comment]
That's just saying it cannot find the package (since it is optional) when it does a version check. 

You can ignore the warning. Although I guess I can not have the warning at all for that part 🤔

**Comment by hoffor** [comment]
a bit confused.. should i install the bepinex package before the unity project patcher package? not sure what the difference is

**Comment by nomnomab** [comment]
If you need it, then you install it after. If you do not need it, then you do not need to install it at all.

The bepinex package is for cases where people want to test existing bepinex plugins in the editor, or play with a custom plugin in-editor.

**Comment by hoffor** [comment]
got it, thank you.. lastly i get this after bepinex package install:
```
Library\PackageCache\com.nomnom.unity-project-patcher-bepinex@21c5857667\Editor\BepInExUtility.cs(51,43): error CS0117: 'BepInExPreloader' does not contain a definition for 'GetBepInExUserSettings'
```
unsure if important. total unity noob so i apologize if these are things i should already understand

**Comment by nomnomab** [comment]
Ah did you install it manually? The window has a button to install it, and it adds the flag it needs (the readme noted this haha).

Since you did it through (I assume) the git way, you'll need to enable the flag here: https://github.com/nomnomab/unity-project-patcher-bepinex/tree/master?tab=readme-ov-file#enabling-this-package

I do need to fix that error so it doesn't happen though in this situation, so thanks for finding it 👍

**Comment by hoffor** [comment]
thank you for the help even though it strayed from the issue a bit. can be hard to parse all the info between the wrapper, patcher and bepinex readmes.. restarted the project and did it all the right way: 1. install unity project patcher using git in package manager, 2. install wrapper same way, 3. install bepinex through the tool window, 4. patch the project, say yes to the input backend thing and wait through the many restarts. all seems successful and functional

**Comment by nomnomab** [comment]
I'm glad that it works! Also, no worries, I'm also glad you brought it up 👍
